File: cf/933/933A.cpp
#include <iostream>
#include <vector>
using namespace std;

int main(){
    int t;
    cin >> t;
    while(t--){
        int n, m, k;
        cin >> n >> m >> k;
        vector<int>v1(n);
        vector<int>v2(m);
        for(int z=0;z<n;z++){
            cin >> v1[z];
        }
        for(int k=0;k<m;k++){
            cin >> v2[k];
        }
        int count=0;
        for(int i=0;i<n;i++){
            for(int j=0;j<m;j++){
                if(v1[i]+v2[j]<=k){
                    count++;
                }
            }
        }
        cout << count << "\n";
    }
}

File: cf/933/933A.py
t = int(input())

for _ in range(t):
    n, m, k = map(int, input().split())
    v1 = list(map(int, input().split()))
    v2 = list(map(int, input().split()))

    v1.sort()
    v2.sort()

    count = 0
    idx1 = 0
    idx2 = 0

    while idx1 < n and idx2 < m:
        if v1[idx1] + v2[idx2] < k:
            count += 1
            if idx1 + 1 < n and (idx2 + 1 >= m or v1[idx1 + 1] - v1[idx1] > v2[idx2 + 1] - v2[idx2]):
                idx1 += 1
            else:
                idx2 += 1
        else:
            break

    print(count)
